engine/src/main/java/org/hibernate/search/engine/cfg/impl/OverriddenConfigurationPropertySource.java

```diff
@@ -47,6 +49,26 @@ public Optional<String> resolve(String key) {
 		}
 	}
 
+	@Override
+	public ScopedConfigurationPropertySource withScope(BeanResolver beanResolver, String namespace, String name) {
+		ConfigurationPropertySource scopedMain = main;
+		ConfigurationPropertySource scopedOverride = override;
+		if ( main instanceof ScopedConfigurationPropertySource ) {
+			scopedMain = ( (ScopedConfigurationPropertySource) main ).withScope( beanResolver, namespace, name );
+		}
+		if ( override instanceof ScopedConfigurationPropertySource ) {
+			scopedOverride = ( (ScopedConfigurationPropertySource) override ).withScope( beanResolver, namespace, name );
+		}
+
+		if ( scopedMain != main || scopedOverride != override ) {
+			return new OverriddenConfigurationPropertySource(
+					scopedMain,
+					scopedOverride
+			);
+		}
+		return this;
```

Comment:
If you're going to implement `withScope` on every `ConfigurationPropertySource` subclass and default to `return this` everywhere, you might as well get rid of `ScopedConfigurationPropertySource` and declare `withScope` directly in `ConfigurationPropertySource`, with a default implementation returning `this`. At least that way you wouldn't have to use `instanceof` everywhere... :zipper_mouth_face: 

Though this looks more complicated than I expected, I need to understand this `withScope` method before I can tell for sure...